cdap-api/src/main/java/co/cask/cdap/api/data/batch/Input.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.data.batch;
+
+import co.cask.cdap.api.data.stream.StreamBatchReadable;
+
+import java.net.URI;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Defines input to a MapReduce job.
+ */
+public abstract class Input {
+
+  private final String name;
+
+  private Input(String name) {
+    this.name = name;
+  }
+
+  /**
+   * Returns an Input defined by a dataset.
+   *
+   * @param datasetName the name of the input dataset
+   */
+  public static Input of(String datasetName) {
+    return of(datasetName, Collections.<String, String>emptyMap());
+  }
+
+  /**
+   * Returns an Input defined by a dataset.
+   *
+   * @param datasetName the name of the input dataset
+   * @param arguments the arguments to use when instantiating the dataset
+   */
+  public static Input of(String datasetName, Map<String, String> arguments) {
```

Comment:
Deprecated StreamBatchReadable. Instead, users should use `Input.ofStream(...)`.
Internally, for now, it will still use StreamBatchReadable, but at least we get the users to not depend on it anymore.